tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -124,6 +124,38 @@ static int setup(void)
 	return TC_PASS;
 }
 
+/* Verify pin-number validation on all non-deprecated API that does
+ * this without __ASSERT.
```

Comment:
All the functions use ASSERT now, the testcase can be removed (or we should write a testcase that tests the ASSERT).

Reply:
Then that's wrong: the plan was to return `-EINVAL` when an invalid pin was provided to a function, per the decision recorded in the PR description.  We should either revert some of those changes to `ASSERT` or get agreement that it's OK to use only assert.  To discuss next API meeting.

Reply:
Per API meeting: test will be removed.

Reply:
done